src/modules/flight_mode_manager/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -145,12 +145,24 @@ void FlightTaskAutoLineSmoothVel::_generateSetpoints()
 void FlightTaskAutoLineSmoothVel::_checkEmergencyBraking()
 {
 	if (!_is_emergency_braking_active) {
-		if (_position_smoothing.getCurrentVelocityZ() > (2.f * _param_mpc_z_vel_max_dn.get())) {
+		// activate emergency braking if significantly outside of velocity bounds
+		if (_position_smoothing.getCurrentVelocityZ() > (1.3f * _param_mpc_z_vel_max_dn.get())
+		    || _position_smoothing.getCurrentVelocityZ() < -(1.3f * _param_mpc_z_vel_max_up.get())
+		    || fabsf(_position_smoothing.getCurrentVelocityX()) > (1.3f * _param_mpc_xy_cruise.get())
+		    || fabsf(_position_smoothing.getCurrentVelocityY()) > (1.3f * _param_mpc_xy_cruise.get())) {
 			_is_emergency_braking_active = true;
 		}
 
 	} else {
-		if (fabsf(_position_smoothing.getCurrentVelocityZ()) < 0.01f) {
+		// deactivate emergency braking if all velocities and accelerations are again within their bounds
+		if (_position_smoothing.getCurrentVelocityZ() < _param_mpc_z_vel_max_dn.get()
+		    && _position_smoothing.getCurrentVelocityZ() > -_param_mpc_z_vel_max_up.get()
+		    && _position_smoothing.getCurrentAccelerationZ() > -_param_mpc_acc_up_max.get()
+		    && _position_smoothing.getCurrentAccelerationZ() < _param_mpc_acc_down_max.get()
+		    && fabsf(_position_smoothing.getCurrentVelocityX()) < _param_mpc_xy_cruise.get()
+		    && fabsf(_position_smoothing.getCurrentVelocityY()) < _param_mpc_xy_cruise.get()
```

Comment:
same here, add `getCurrentVelocityXY` and `getCurrentAccelerationX` that return a `Vector2f` and use `longerThan(`